wasm-chain-integration/src/lib.rs

```diff
@@ -323,12 +334,19 @@ impl<'a> HasCommon for InitHost<'a> {
 
     fn param(&self) -> &[u8] { &self.param }
 
-    fn metadata(&self) -> &ChainMetadata { &self.init_ctx.metadata }
+    fn metadata(&self) -> &Self::MetadataType { self.init_ctx.metadata() }
 
-    fn policies_bytes(&self) -> &[u8] { &self.init_ctx.sender_policies }
+    fn policies(&self) -> ExecResult<&Self::PolicyType> { self.init_ctx.sender_policies() }
 }
 
-impl<'a> HasCommon for ReceiveHost<'a> {
+impl<'a, Ctx> HasCommon for ReceiveHost<'a, Ctx>
+where
+    Ctx: HasReceiveContext,
```

Comment:
This is a minor stylistic issue, but I think we generally try to use 
```rust
impl<'a, Ctx: HasReceiveContext> ...
```
instead of the `where` clause, when the constraint is a simple one like this.